src/preamble.js

```diff
@@ -356,6 +356,11 @@ var __ATPOSTRUN__ = []; // functions called after the main() is called
 
 var runtimeInitialized = false;
 var runtimeExited = false;
+var runtimeKeepaliveCounter = 0;
+
+function keepRuntimeAlive() {
+  return noExitRuntime || runtimeKeepaliveCounter > 0;
+}
```

Comment:
In general we prefer to put stuff in JS libraries, instead of always including it in preamble/postamble. Why does this do the opposite?

Reply:
Since this is now being called from shell.js and basically always required I thought it is simpler in this particular case to add this tiny function to the preamble.js.    It also makes sense to me to have runtimeKeepaliveCounter  declared alongside the existing noExitRuntime since they work together.   Do you think I should revert this part of the change?

Reply:
The other reason for putting stuff in the library rather than preamble/postamble is so that it can be shared between MINIMAL_RUNTIME and the normal runtime (which have different preamble files), but MINIMAL_RUNTIME doesn't want this function so it makes less sense in the library (unlike for example `establishStackSpace` and `invokeEntryPoint` which are also force includes from the library).

Reply:
I guess that's fair, yeah, as this is something not relevant for minimal runtime anyhow, and the code gets simpler, sgtm.